### DISCUSSION TITLE
Module mikrotik:error in unmarshalResponse: expected a sequence when unmarshalling a VB, got 2

### DISCUSSION THREAD

**Discussion by matgapa** [prometheus/snmp_exporter#1205] [Q&A]
### Host operating system: output of `uname -a`
Ubuntu 20.04

### snmp_exporter version: output of `snmp_exporter -version`

snmp_exporter, version 0.22.0 

--------------
 I have two router mikrotik I want use prometheus with grafana for monitoring panel... I tried with several version 0.17 to 0.25 only worked 0.22 correctly with panel on grafana this no really important only want solve problem with snmp_exporter this running on port 9116 when try  use module mikrotik on 1 router  this working without problem but another router error 

working:
http://xxx.xxx.xxx.xxx:9116/snmp?auth=public_v2&module=mikrotik&target=xxx.xxx.xxx.aaa

show all metrics

not working 

http://xxx.xxx.xxx.xxx:9116/snmp?auth=public_v2&module=mikrotik&target=xxx.xxx.xxx.bbb

An error has occurred while serving metrics:

error collecting metric Desc{fqName: "snmp_error", help: "Error scraping target", constLabels: {}, variableLabels: []}: error walking target xxx.xxx.xxx.bbb: error in unmarshalResponse: expected a sequence when unmarshalling a VB, got 25

id use another module "if_mib" show some dates but need use module mikrotik
-----

from shell using snmpwalk all working fine

snmpwalk -v2c -c public xxx.xxx.xxx.aaa

show metrics

snmpwalk -v2c -c public xxx.xxx.xxx.bbb 

show metrics

**Comment by HaradaKumiko** [comment]
any update for this issue?

**Comment by faizalabroni** [comment]
I have the same issue. Any solution?

**Reply by HaradaKumiko** [reply]
belum ada juga bro wkwkw

**Comment by izzais** [comment]
I fixed this by removing/comment the `modules: mikrotik: walk:` in snmp.yml 
```
      - 1.3.6.1.4.1.14988
```
and replace it to 
```
      - 1.3.6.1.4.1.14988.1.1.1
      - 1.3.6.1.4.1.14988.1.1.2
      - 1.3.6.1.4.1.14988.1.1.3
      - 1.3.6.1.4.1.14988.1.1.4
      - 1.3.6.1.4.1.14988.1.1.5
      - 1.3.6.1.4.1.14988.1.1.6
      - 1.3.6.1.4.1.14988.1.1.7
```

**Reply by SuperQ** [reply]
Care to make a PR adjusting the generator.yml to select more specific OIDs?
